plugin-server/tests/helpers/clickhouse.ts

```diff
@@ -30,26 +30,29 @@ export async function resetTestDatabaseClickhouse(extraServerConfig?: Partial<Pl
     ])
 }
 
-export async function delayUntilEventIngested(
-    fetchEvents: () => Promise<any[] | number>,
-    minCount = 1,
+export async function delayUntilEventIngested<T extends any[] | number>(
+    fetchData: () => Promise<T>,
+    minLength = 1,
     delayMs = 100,
     maxDelayCount = 100
-): Promise<void> {
+): Promise<T> {
     const timer = performance.now()
+    let data: T
+    let dataLength = 0
     for (let i = 0; i < maxDelayCount; i++) {
-        const events = await fetchEvents()
-        const eventCount = typeof events === 'number' ? events : events.length
+        data = await fetchData()
+        dataLength = typeof data === 'number' ? data : data.length
         if (determineNodeEnv() === NodeEnv.Development) {
             console.log(
-                `Waiting. ${Math.round((performance.now() - timer) / 100) / 10}s since the start. ${eventCount} event${
-                    eventCount !== 1 ? 's' : ''
+                `Waiting. ${Math.round((performance.now() - timer) / 100) / 10}s since the start. ${dataLength} event${
+                    dataLength !== 1 ? 's' : ''
                 }.`
             )
         }
-        if (eventCount >= minCount) {
-            return
+        if (dataLength >= minLength) {
+            return data
         }
         await delay(delayMs)
     }
+    return data
```

Comment:
Suggestion: Let's not do this - this helper is getting too clever IMO.

Yes, we will have small bits of repetition but having overtly smart utils is a smell.

Reply:
I kind of see zero point in _always_ having to do:
```JS
await delayUntilEventIngested(() => db.fetchSomeData())
const data = await db.fetchSomeData()
```
This pattern is so constant and the refactor so simple, that this seems pretty much obvious, even if not ideologically pure